### ISSUE TITLE
FireFox Issue: Some code of a Polymer.dart application doesn't work in FireFox (but works in other browsers!).

### ISSUE THREAD

**Issue by DartBot** [dart-lang/polymer-dart#245]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [RWander](https://github.com/RWander)**
_Originally opened as dart-lang/sdk#20141_

----

**Attachment:**
[cc.zip](https://storage.googleapis.com/google-code-attachments/dart/issue-20141/comment-0/cc.zip) (198.16 KB)

**Comment by DartBot** [comment]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [RWander](https://github.com/RWander)**

----

**Attachment:**
[cc.zip](https://storage.googleapis.com/google-code-attachments/dart/issue-20141/comment-1/cc.zip) (198.20 KB)

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I think this is related to https://github.com/Polymer/platform-dev/issues/32
(https://github.com/Polymer/ShadowDOM/issues/328 is similar but seems to be another problem)

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

Ctrl+Shift+j showed the error: 
TypeError: Argument 1 of Window.getDefaultComputedStyle does not implement interface Element.
after loading the entry page of your example.

**Comment by DartBot** [comment]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [RWander](https://github.com/RWander)**

----

Note that the attached sample works fine if cc.html page don't use polymer.dart and dependency (dart.js, platform.js, dart_support.js)..

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

*Added [Pkg-Polymer](../labels/Pkg-Polymer), [Area-Pkg](../labels/Area-Pkg), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-Medium](../labels/Priority-Medium), [PolymerMilestone-Next](../labels/PolymerMilestone-Next) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [RWander](https://github.com/RWander)**

----

Still doesn't work in Polymer 0.13.0 yet..

**Comment by DartBot** [comment]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [RWander](https://github.com/RWander)**

----

please see https://code.google.com/p/dart/issues/detail?id=20725

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

Ok, I spent some time on this and discovered the actual bug. Pull request out to  https://github.com/Polymer/ShadowDOM/pull/500.

I am going to work on patching this as well and releasing it as a +1 so that you can get unblocked without having to wait on the next migration.
______
*Set owner to @jakemac53.*
*Added [Started](../labels/Started) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

Issue dart-lang/sdk#20725 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

Change cherrypicked in dart-lang/sdk@3c81cd3ae3541c6aad7612649343fe29438ca2e6, assuming the bots stay green I will publish this later today or tomorrow morning
______
*Added [Fixed](../labels/Fixed) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

published as 0.6.0+1, run pub upgrade and you should be ready to go

**Comment by DartBot** [comment]
<a href="https://github.com/RWander"><img src="https://avatars.githubusercontent.com/u/2590778?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [RWander](https://github.com/RWander)**

----

Thanks! it works!

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

Yay! Sorry this one blocked you for so long, feel free to put friendly pings on these issues if it seems like its falling through the cracks. Also, its usually possible for us to cherrypick individual fixes if they are causing large headaches as opposed to waiting for the next major version so feel free to request that as needed in the future, and we will do our best to help out.
